source/common/mongo/proxy.cc

```diff
@@ -226,9 +235,13 @@ void ProxyFilter::onEvent(Network::ConnectionEvent event) {
 }
 
 Network::FilterStatus ProxyFilter::onData(Buffer::Instance& data) {
+  if (delay_timer_)
```

Comment:
if we have an active delay here, won't all requests be blocked? ie non query requests are being delayed.

Reply:
I think we could not do on a per operation delays until we make mongo proxy filter full L7 filter, and not the bump in the wire + tcp proxy.

There is still a problem with capturing stats in here, i'll fix.